napari/layers/utils/layer_utils.py

```diff
@@ -204,81 +216,485 @@ def _nanmax(array):
+    | |0| |1| |2| |
+    ---------------
+    | |3| |4| |5| |
+    ---------------
+    | |6| |7| |8| |
+    ---------------
+
+    Parameters
+    ----------
+    shape: Sequence[int]
+        Either the shape of the raw data (in pixels) or the block shape of the data (chunks)
+    plane_indices: Sequence[Sequence[int]]
+        Bottom, middle and top plane or single plane index for each non-visible dimension.
+    offset: int
+        Number of visible dimensions.
+    chunk_shape: None | Tuple[int]
+        The size per dimension of the chunks.
+
+    Returns
+    -------
+    slices: Union[None, List[Tuple[Union[int, slice], ...]]]
+        A list of crop slices.
+    """
+    plane_shape = shape[-offset:]
+
+    (
+        max_chunks_per_plane,
+        max_allowed_chunks,
+        chunk_size_y,
+        chunk_size_x,
+    ) = _calculate_chunk_parameters(plane_indices, offset, chunk_shape)
+
+    plane_indices, y_start_indices, x_start_indices = _get_pixel_start_indices(
+        plane_indices,
+        plane_shape,
+        offset,
+        chunk_size_y,
+        chunk_size_x,
+        chunk_shape,
+    )
+
+    start_indices = [
+        (y_start, x_start)
+        for y_start in y_start_indices
+        for x_start in x_start_indices
+    ]
+    num_start_indices = len(start_indices)
+    chunk_multiplier = min(
+        max(max_chunks_per_plane // num_start_indices - 1, 0),
+        MAX_NUMBER_OF_CHUNKS,
+    )
+
+    # We have at least 1 chunk per plane, but not all chunks are allowed to be loaded. Only lazy data
+    indices: Tuple[int, ...]
+    if num_start_indices > max_chunks_per_plane >= 1:
+        if num_start_indices == 3:
+            # center chunk
+            indices = (1,)
+        # Below num_start_indices can only be 9
+        else:
+            if max_chunks_per_plane // 5 == 1:
+                # + pattern of chunks, check function doc string
+                indices = (1, 3, 4, 5, 7)
+            elif max_chunks_per_plane // 3 == 1:
+                # horizontal dashed line of chunks, check function doc string
+                indices = (3, 4, 5)
+            else:
+                # center chunk, check function doc string
+                indices = (4,)
+        start_indices = [start_indices[i] for i in indices]
+    elif max_chunks_per_plane < 1:
+        index = 1 if num_start_indices == 3 else 5
+        start_indices = [start_indices[index]]
+        # get the middle plane if 3 planes exist, otherwise first plane.
+        plane_indices = [plane_indices[max(1, len(plane_indices) - 1)]]
+
+    slices_y = _get_slices(
+        start_indices, len(y_start_indices), chunk_size_y, 0, chunk_multiplier
+    )
+    slices_x = _get_slices(
+        start_indices, len(x_start_indices), chunk_size_x, 1, chunk_multiplier
+    )
+    plane_slices: List[Tuple[Union[slice, int], ...]] = list(
+        zip(slices_y, slices_x)
+    )
+    if len(plane_indices) == 0 or len(plane_indices[0]) == 0:
+        return plane_slices
```

Comment:
```suggestion
        return None
```

If it is expected to be None then we should explicitly return None (from documentation)

Reply:
resolved with updated type hint